src/v/kafka/server/handlers/topics/types.h

```diff
@@ -156,4 +158,54 @@ std::vector<kafka::creatable_topic_configs> report_topic_configs(
   const cluster::metadata_cache& metadata_cache,
   const cluster::topic_properties& topic_properties);
 
+// Either parse configuration or return nullopt
+inline std::optional<bool>
+get_bool_value(const config_map_t& config, std::string_view key) {
+    if (auto it = config.find(key); it != config.end()) {
+        try {
+            // Ignore case.
+            auto str_value = it->second;
+            std::transform(
+              str_value.begin(),
+              str_value.end(),
+              str_value.begin(),
+              [](const auto& c) { return std::tolower(c); });
+            return string_switch<std::optional<bool>>(str_value)
+              .match("true", true)
+              .match("false", false);
+        } catch (const std::runtime_error&) {
+            throw boost::bad_lexical_cast();
+        };
+    }
+    return std::nullopt;
+}
+
+inline model::shadow_indexing_mode
+get_shadow_indexing_mode(const config_map_t& config) {
```

Comment:
did these need to move to the header or could we have added a declaration to the header? or alternatively, could the implementation of delete_retention_ms_validator have been in the same .cc file?

Reply:
the general rule of thumb is to be reducing the amount of code in headers. its not any sort of hard and fast rule, but i don't see in this case why it needs to move to the header. is there a dependency issue that makes it hard to do?

Reply:
Moved declaration/implementation